### PR TITLE
579498 filter dialog types

### PR DESCRIPTION
**Pull request by eobrienPilz**
filterDialog
ProblemView

We have a customized problem view for our application.
It displays markers of our own type pilzproblemmarker

i.e.

Defined like this.
`
       <extension point="org.eclipse.core.resources.markers"
        id="pilzproblemmarker"
		name="pilz problem marker">
        	<super
                type="org.eclipse.core.resources.problemmarker">
          	</super>
		.......
`

We associate it with the problem view like this.
`
      <extension point="org.eclipse.ui.ide.markerSupport">
      	id="com.pilz.pas.platform.gui.pilzproblemmarker"/>
 `

We create our markers like this with our custom marker as a super.
`    <extension point="org.eclipse.core.resources.markers"
          id="com.pilz.pas.hal.errorhandling.ErrorCodes.STATEMENT_INDEX_OUT_OF_BOUNDS"
          name="%SUPPRESSION_MSG_STATEMENT_INDEX_OUT_OF_BOUNDS">
          <super
                type="com.pilz.pas.platform.gui.pilzproblemmarker">
          </super>
		.......`

This works well with the filter dialog. 
1) All our markers get displayed in the type section of the filter dialog.
2) Other problem markers that we don't care about (e.g. Java problems) are not displayed.

Only issue is that the supertype pilzproblemmarker is shown : see the screenshot
We need to hide it.

### CODE REVIEW

**Comment by eobrienPilz** [comment]
Link to the bugzilla issue: https://bugs.eclipse.org/bugs/show_bug.cgi?id=579498

I just added a comment there describing the patch proposal

**Review by jukzi — CHANGES_REQUESTED** [review]
The version in the Manifest needs to be updated.

16:42:27.335 [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:3.0.0-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.ui.ide: Only qualifier changed for (org.eclipse.ui.ide/3.19.0.v20220610-1747). Expected to have bigger x.y.z than what is available in baseline (3.19.0.v20220511-1638) -> [Help 1]

* I do not understand the issue - could you provide screenshots before/after?
* new files need a copyright in the header
* please avoid newline changes.

API

**Comment by eobrienPilz** [comment]
> The version in the Manifest needs to be updated.
> 
> 16:42:27.335 [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:3.0.0-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.ui.ide: Only qualifier changed for (org.eclipse.ui.ide/3.19.0.v20220610-1747). Expected to have bigger x.y.z than what is available in baseline (3.19.0.v20220511-1638) -> [Help 1]
> 
> * I do not understand the issue - could you provide screenshots before/after?
> * new files need a copyright in the header
> * please avoid newline changes.
> 
> API

Are there guidelines somewhere on what the version should be ?

Should I change from 
Bundle-Version: 3.19.0.qualifier 
to 
Bundle-Version: 3.19.1.qualifier

**Comment by jukzi** [comment]
> Bundle-Version: 3.19.1.qualifier

we increase by 100. see https://wiki.eclipse.org/Version_Numbering#When_to_change_the_service_segment
3.19.100

File: tests/org.eclipse.ui.tests/plugin.xml
Comment (jukzi) [inline tests/org.eclipse.ui.tests/plugin.xml]: please avoid to change lines that do not belong to the change.

File: bundles/org.eclipse.ui.ide/schema/markerSupport.exsd
Comment (jukzi) [inline bundles/org.eclipse.ui.ide/schema/markerSupport.exsd:198]: Is the attribute optional? What is the default value that is used when the attribute is not set? please document.
This is probably an API change, which requires multiple committers to agree upon. good luck.

**Review by jukzi — CHANGES_REQUESTED** [review]

**Comment by eobrienPilz** [comment]
> > Bundle-Version: 3.19.1.qualifier
> 
> we increase by 100. see https://wiki.eclipse.org/Version_Numbering#When_to_change_the_service_segment 3.19.100

Reading the wiki its still not clear if it should be a minor change or a service change. The extension point has changed so I thought maybe it should be minor. However if I increase minor to 3.20.0 I get an error from API Analysis in eclipse saying there is no API change.

**Comment by jukzi** [comment]
> Reading the wiki its still not clear if it should be a minor change or a service change. The extension point has changed so I thought maybe it should be minor. However if I increase minor to 3.20.0 I get an error from API Analysis in eclipse saying there is no API change.

It's not a binary break that the API Analysis detects. However it is an API change. Some guru can force an increase to 3.20 but let's wait if others agree about that change.

File: bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java
Comment (jukzi) [inline bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java:208]: i wonder that the default does change the previous implementation. please explain.

File: tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/SubTypeOnlyTestView.java
Comment (jukzi) [inline tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/SubTypeOnlyTestView.java:1]: copyright should be first lines in file

File: tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/SubTypeOnlyTestView.java
Comment (jukzi) [inline tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/SubTypeOnlyTestView.java]:
```suggestion
 * Copyright (c) 2022 YOUR NAME and others.
```

File: bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java
Comment (eobrienPilz) [reply bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java:208]: If you look at the change in ContentGeneratorDescriptor.getMarkerTypes you see that the previous implementation was populating types and subtypes. With the change the application attribute is used to determine what to include. If there is no application attribute then the type and subtype are used just like the old implementation.

... so yes the implementation is changed but no change in behavior for existing clients.

Comment (jukzi) [reply bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java:208]: you changed the order of addAll() and add(). probably not a problem - but why? prefer to keep the old code if there is no need to change

Comment (eobrienPilz) [reply bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java:208]: No reason for the order change. Its swapped back now.

**Review by jukzi — APPROVED** [review]
LGTM (did not test it though) - @mickaelistria API change needs more reviewers

File: tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/NoApplicationAttribTestView.java
Comment (jukzi) [inline tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/NoApplicationAttribTestView.java]: no `@since` in tests, please

File: tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/TypeAndSubTypeTestView.java
Comment (jukzi) [inline tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/TypeAndSubTypeTestView.java]: Regarding all files: you may replace "Contributor" with your own name or company.

**Review by jukzi — COMMENTED** [review]
Overall it still looks good to me. May be you can just apply my last comments and then this is good to merge if nobody insists.

File: tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerTypeTests.java
Comment (jukzi) [inline tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerTypeTests.java:35]: you need to add the Test to a suite, so that it is executed on CI

Comment (jukzi) [reply tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerTypeTests.java:35]: https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-144/15/testReport/UiTestSuite%20org.eclipse.ui.tests.internal.InternalTestSuite%20org.eclipse.ui.tests.markers/MarkerTypeTests/

**Pull request merged**
